corehq/apps/ota/models.py

```diff
@@ -89,3 +90,40 @@ def _delete_restore_blob(self):
         self.restore_blob_id = None
 
         return deleted
+
+
+class SerialIdBucket(models.Model):
+    """
+    Model used to keep track of an incrementing, unique integer
+    to be used in serial ID generation
+    """
+    domain = models.CharField(max_length=255)
+    bucket_id = models.CharField(max_length=255)
+    current_value = models.IntegerField(default=-1)
+
+    class Meta:
+        index_together = ('domain', 'bucket_id',)
+        unique_together = ('domain', 'bucket_id',)
+
+    @classmethod
+    def get_next(cls, domain, bucket_id, session_id=None):
+        if session_id is not None:
+            return cls._get_next_cached(domain, bucket_id, session_id)
+        else:
+            return cls._get_next(domain, bucket_id)
+
+    @classmethod
+    @quickcache(['domain', 'bucket_id', 'session_id'])
```

Comment:
Is there is a way to combine this with `_get_next` and use `quickcache` with `skip_args` instead? 
Absolutely fine if you want to intentionally keep this separate to make it clearer.

Reply:
The problem is that it'll still write to the cache if session id isn't provided, which is useless, so this approach just circumvents the cache altogether.  It felt like the clearest way to do that.